public/script.js

```diff
@@ -6233,8 +6237,27 @@ export function setExternalAbortController(controller) {
     abortController = controller;
 }
 
+/**
+ * Sets a character array index.
+ * @param {number|string|undefined} value
+ */
 export function setCharacterId(value) {
-    this_chid = value;
+    switch (typeof value) {
+        case 'bigint':
+        case 'number':
+            this_chid = String(value);
+            break;
+        case 'string':
+            this_chid = !isNaN(parseInt(value)) ? value : undefined;
+            break;
+        case 'object':
```

Comment:
Object to undefined? What's the reasoning behind this?
Where would this happen and be relevant, and not log the console error?

**If** we wanna do something about char objects here, it would make more sense to check if `avatar` field is present, and then search for that in the char array.

Reply:
Hmmh. I mean it's okay. Still feel like every other object shouldn't *work* and log an error, not unsetting the char.
But we can keep it.